lib/irb/context.rb

```diff
@@ -555,9 +555,12 @@ def evaluate(line, line_no) # :nodoc:
         IRB.set_measure_callback
       end
 
+      # We need to use differente path to distinguish source_location of method defined in the actual file and method defined in irb session.
+      eval_path = File.exist?(irb_path) ? "#{irb_path}(#{IRB.conf[:IRB_NAME]})" : irb_path
```

Comment:
Can this be an ivar for `Context` instead or a memoised method so we don't run `File.exist?` on every input? It'd mean `IRB.conf[:IRB_NAME]` changes between inputs won't be picked up, but IMO that's ok.

Reply:
IRB_NAME does not change frequently, but irb_path does. (in irb_load and rdbg)
Updated to memoize irb_path existence considering irb_path change.